kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ingress/ProxyIngressLayout.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.ingress;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.proxy.config.VirtualClusterGateway;
+
+import static io.kroxylicious.kubernetes.operator.ProxyDeployment.PROXY_PORT_START;
+
+public record ProxyIngressLayout(List<VirtualClusterLayout> clusterLayouts) {
```

Comment:
Layout is the thing responsible for allocating proxy ports to cluster+ingress

Reply:
It's its responsible for allocatoring ports then why not `ProxyPortAllocator`?

Reply:
This was split into model and allocator